pkg/stanza/operator/input/windows/input.go

```diff
@@ -232,88 +233,72 @@ func (i *Input) read(ctx context.Context) int {
 
 // processEvent will process and send an event retrieved from windows event log.
 func (i *Input) processEvent(ctx context.Context, event Event) {
-	var providerName string // The provider name is only retrieved if needed.
-	if !i.raw || len(i.excludeProviders) > 0 {
-		var err error
-		providerName, err = event.GetPublisherName(i.buffer)
-		if err != nil {
-			i.Logger().Error("Failed to get provider name", zap.Error(err))
-			return
-		}
+	providerName, err := event.GetPublisherName(i.buffer)
+	if err != nil {
+		i.Logger().Error("Failed to get provider name", zap.Error(err))
+		return
 	}
-
-	if len(i.excludeProviders) > 0 {
-		for _, excludeProvider := range i.excludeProviders {
-			if providerName == excludeProvider {
-				return
-			}
-		}
+	if _, exclude := i.excludeProviders[providerName]; exclude {
+		return
 	}
 
-	if i.raw {
-		rawEvent, err := event.RenderSimple(i.buffer)
-		if err != nil {
-			i.Logger().Error("Failed to render raw event", zap.Error(err))
+	if i.supressRenderingInfo {
+		simpleEvent, simpleErr := event.RenderSimple(i.buffer)
+		if simpleErr != nil {
+			i.Logger().Error("Failed to render simple event", zap.Error(simpleErr))
 			return
 		}
-
-		i.sendEventRaw(ctx, rawEvent)
+		i.sendEvent(ctx, simpleEvent)
 		return
 	}
 
 	publisher, openPublisherErr := i.publisherCache.get(providerName)
 	if openPublisherErr != nil {
-		// This happens only the first time the code fails to open the publisher.
+		// Do not return. Log error here and try to send as simple event later.
 		i.Logger().Warn(
 			"Failed to open event source, respective log entries cannot be formatted",
 			zap.String("provider", providerName), zap.Error(openPublisherErr))
-	}
-
-	if publisher.Valid() {
-		formattedEvent, err := event.RenderFormatted(i.buffer, publisher)
-		if err == nil {
-			i.sendEvent(ctx, formattedEvent)
+	} else if publisher.Valid() {
```

Comment:
The `else if` will cause the information of the very first event from a publisher to not be recorded if there is an error opening the publisher.

Reply:
I don't understand this. What do you suggest should change here?

Reply:
You handled it via the refactoring.